### PR TITLE
Build Calendar in React Using React-Calendar Library

### LINKED ISSUE #5832: [LanguageBuild Calendar in React Using React-Calendar Library
### Topic Suggestion
Build Calendar in React Using React-Calendar Library 

#### Proposed title of article
Build Calendar in React Using React-Calendar Library 

Use keyword research to improve your article's SEO.

#### Proposed article introduction
React-Calendar is a simple calendar library that provides the ability to pick days, months, years, or even decades. It also supports date range selection and a variety of languages for more complex use cases.

A React calendar component is a good choice when you need to allow users to select a specific date, like if your app has a form field for date of birth or employee start date. Another good example would be an application that allows users to book appointments. A user would be able to select a specific date and time slot from available options. For scenarios like this, a calendar component is probably the best bet.

It’s a very flexible and versatile library that you can use in virtually any application.

#### Key takeaways 
The tutorial will show readers how to:

1. Integrate react-calender library into a react application.
2. How to enable date range feature using selectRange.
3. How to style and customize react-calendar
4. 
#### Article quality
The article will give a detailed guide into react-calendar library. It will explain key concepts to readers which is unavailable in the documentations and other platforms.  

#### References
N/a

### Conclusion
At the end of the tutorial, the reader should be able to achieve the goals listed above while also learning how to create useful calendars in their applications.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by codeInn001**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5832

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by jumamiller** [comment]
A great article, looking forward to interacting with you as soon as possible!

Regards,

File: content/articles/build-react-calendar-library/index.md [hunk 1]
```diff
@@ -0,0 +1,362 @@
+# Build a Calendar in React Using React-Calendar Library
+
+![hero](/engineering-education/build-react-calendar-library/hero.jpg)
```
Comment (jumamiller) [inline content/articles/build-react-calendar-library/index.md:3]: Kindly follow the article formats, I can see you have published quite a number, kindly check on the intro part!

File: content/articles/build-react-calendar-library/index.md [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+
+Oftentimes, you want to manage and manipulate dates in your web app. These may include adding events or setting reminders. Having a calendar in your web application is a way to implement this. 
+
+In this tutorial, you will learn how to manage and manipulate dates in React using React-Calendar.
+
+### What is React-Calendar
+React-calendar is an easy-to-use react calendar library. It provides functionality that helps you to manipulate and manage date and time. 
+
```
Comment (jumamiller) [inline content/articles/build-react-calendar-library/index.md:11]: Are some paragraphs incomplete?

File: content/articles/build-react-calendar-library/index.md [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+### Goals
+
+In this tutorial, you will learn:
+* How to add React-Calendar to your React app.
+* How to style React-Calendar.
+* How to customize React-Calendar.
+* How to select a date range in React-Calendar.
+* How to create a date picker
+
+### Prerequisite
```
Comment (jumamiller) [inline content/articles/build-react-calendar-library/index.md:22]: plurals?

File: content/articles/build-react-calendar-library/index.md [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+2. Clear the boilerplate.
+
+3. Import calendar component from react-calendar in `App.js`.
+
+4. Copy the code below into your `App.js` file.
+
+
+
+**src/App.js**
+```jsx
```
Comment (jumamiller) [inline content/articles/build-react-calendar-library/index.md:95]: is it supported by markdown?

File: content/articles/build-react-calendar-library/index.md [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+export default App;
+```
+
+In the code above, we used `useState` to store a date and we set the present date as its initial value using `Javascript Date object`. We then pass the date as a value to our Calendar component. 
+
+When a user selects a date the value of date will be set to the user's selected date. The selected date is then printed below your calendar.
+
+Your application should look like the one below.
+
+![react-application-structure](react-calendar-default.png)
```
Comment (jumamiller) [inline content/articles/build-react-calendar-library/index.md:128]: Update these image links!

File: content/articles/build-react-calendar-library/index.md [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+
+### Build Booking App with React-Calendar
+You can integrate other features with react-calendar to build an application with a complex feature.
+
+We will demonstrate this by building a booking app. In the booking app, users can click on a date of their choice. After clicking on the desired date, the application will display the available time slot. Users can then select a time slot. After selecting a time slot, the date and time of appointment will be displayed on the screen.
+
+Copy the code below in your `app.js` file.
+**src/App.js**
+
+```jsx
```
Comment (jumamiller) [inline content/articles/build-react-calendar-library/index.md:217]: Once again, doesn't seem to be supported! Have a look

File: content/articles/build-react-calendar-library/index.md [hunk 7]
```diff
@@ -0,0 +1,362 @@
...
+We passed in date as a prop from  `App.js` to `Time.js` finally to `Times.js`.
+
+When the value of `info` changes to true, the app displays the time slot and date selected by the user.
+
+Our final application will look like the image below.
+
+![react-application-structure](booking-app-demo.png)
+
+### Conclusion
+We covered how to integrate react-calendar app, customize, set a date range and add a booking feature into our react app.
```
Comment (jumamiller) [inline content/articles/build-react-calendar-library/index.md:359]: Add more content on this section!

**Review by jumamiller — CHANGES_REQUESTED** [review]
Work on the few suggestions above!

File: content/articles/build-react-calendar-library/index.md (hunk 2 above)
Comment (codeInn001) [reply content/articles/build-react-calendar-library/index.md:11]: Please, can you be more explicit?

**Comment by codeInn001** [comment]
@jumamiller Thanks, corrections made.

**Comment by ahmadmardeni1** [comment]
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.


[[Build Calendar in React Using React-Calendar Library](https://github.com/section-engineeringeducation/engineering-education/pull/6868#)
#6868](https://github.com/section-engineering-education/engineering-education/files/8183737/New.Scan.7_05.AM.Copyleaks.Report.pdf)

**Comment by codeInn001** [comment]
> Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept. Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content. Be sure to see our resources page to see more info on plagiarism and what is considered as such.
> 
> [Build Calendar in React Using React-Calendar Library#6868](https://github.com/section-engineering-education/engineering-education/files/8183737/New.Scan.7_05.AM.Copyleaks.Report.pdf)

**Comment by codeInn001** [comment]
> > Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept. Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content. Be sure to see our resources page to see more info on plagiarism and what is considered as such.
> > [Build Calendar in React Using React-Calendar Library#6868](https://github.com/section-engineering-education/engineering-education/files/8183737/New.Scan.7_05.AM.Copyleaks.Report.pdf)

@ahmadmardeni1 Plagiarism corrected. Please, kindly note, Most of the words flagged 
for plagiarism are words common in technical articles.

Here is my report for you to go through. Thanks. 

[3931063-d2a2f89252457139f721-1646699607520.pdf](https://github.com/section-engineering-education/engineering-education/files/8201928/3931063-d2a2f89252457139f721-1646699607520.pdf)

**Comment by codeInn001** [comment]
@jumamiller corrections made.
Thanks

**Review by jumamiller — APPROVED** [review]

**Review by jumamiller — APPROVED** [review]

**Comment by ahmadmardeni1** [comment]
Hey @jumamiller Please whenever an author revisits his/her article, you need to add the "need plagiarism check" label again. This article is ready now, but this is for your future reviews.

**Comment by ahmadmardeni1** [comment]
Great work team!! @codeInn001 @jumamiller

**Pull request merged**